mousestyles/PathFeatures.py

```diff
@@ -0,0 +1,167 @@
+    >>> dist = compute_distances(path)
+	"""
+
+    indices = path_obj.index[:len(path_obj)-1]
+    distances = [np.linalg.norm(path_obj.loc[i,'x':'y'] - path_obj.loc[i+1,'x':'y']) for i in indices]
+    return(distances)
+
+def take_time_diffs(path_obj):
+	r"""
+	Returns a list of the time differences along the path.
+	Each element of the list is the difference of timestamp between the adjacent points in the path.
+	The length of the list is equal to the length of the path minus 1.
+
+	Parameters
+    ----------
+    path_obj : pandas.DataFrame
+        CT, CX, CY coordinates and homebase status.
+
+	Returns
+    -------
+    time differences : list
+    	contains the time differences along the path.
+
+	"""
+
+    indices = path_obj.index[:len(path_obj)-1]
+    time_diffs = [path_obj.loc[i+1, 't'] - path_obj.loc[i, 't'] for i in indices]
+    return(time_diffs)
+
+def compute_speeds(distances, time_diffs):
+	r"""
+	Returns a list of the speeds along the path.
+	Each element of the list is the ratio of the traveled distance to the time difference.
+	The length of the list is equal to the length of distances or time_diffs.
+
+	Parameters
+    ----------
+    distances : list
+        the traveled distances along the path. Expecting the output of compute_distances.
+    time_diffs : list
+    	the time difference within the path. Expecting the output of take_time_diffs. 
+    	Must have the same length as distances. Should not contain 0; otherwise output would 
+    	contain inf.
+
+	Returns
+    -------
+    speeds : list
+    	contains the speeds along the path.
+
+	"""
+
+    if len(distances) is not len(time_diffs):
+        raise ValueError("lengths of distances and time_diffs must be same")
+    
+    ## caution: time_diff should not contain 0.
+    
+    speeds = [d/t for d,t in zip(distances, time_diffs)]
+    return(speeds)
+
+def compute_accelerations(speeds, timestamps):
+	r"""
+	Returns a list of the accelerations along the path.
+	Each element of the list is the ratio of the speed to the time difference.
+	The length of the list is equal to the length of speeds minus 1.
+
+	Parameters
+    ----------
+    speeds : list
+        the traveled distances along the path. Expecting the output of compute_distances.
+    timestamps : list
+    	the time difference within the path. Expecting the output of take_time_diffs. 
+    	Must have the same length as distances. Should not contain 0; otherwise output would 
+    	contain inf.
+
+	Returns
+    -------
+    speeds : list
+    	contains the speeds along the path.
+
+	"""
+
+    if len(speeds) is not len(timestamps)-1:
+        raise ValueError("lengths of speeds must be the length of timestamps minus 1")
+        
+    ## caution: timestamps shoud not contain same time at adjacet rows
+    
+    speeds_diff = [ x - y for x,y in zip(speeds[:len(speeds)], speeds[1:])] 
+    time_diffs = [ x - y for x,y in zip(timestamps[:len(timestamps)-1], timestamps[2:])]
```

Comment:
Similar to above, could use something like this to throw an error when two adjacent times are the same:

`if np.count_nonzero(time_diffs) is not len(time_diffs): 
        raise ValueError("timestamps shoud not contain same time at adjacet rows")`